src/components/Block/SpaceFormProfile.vue

```diff
@@ -16,67 +16,79 @@ const props = withDefaults(
 const emit = defineEmits<{
   (e: 'errors', value: any);
   (e: 'pick', field: any);
+  (e: 'no-network');
 }>();
 
 const availableNetworks = enabledNetworks.map(id => ({
   id,
   name: getNetwork(id).name
 }));
 
-const definition = {
-  type: 'object',
-  title: 'Space',
-  additionalProperties: false,
-  required: ['name', 'walletNetwork', 'walletAddress'],
-  properties: {
-    name: {
-      type: 'string',
-      title: 'Name',
-      minLength: 1,
-      examples: ['Space name']
-    },
-    description: {
-      type: 'string',
-      format: 'long',
-      title: 'About',
-      examples: ['Space description']
-    },
-    externalUrl: {
-      type: 'string',
-      title: 'Website',
-      examples: ['Website URL']
-    },
-    github: {
-      type: 'string',
-      title: 'GitHub',
-      examples: ['GitHub handle']
-    },
-    twitter: {
-      type: 'string',
-      title: 'Twitter',
-      examples: ['Twitter handle']
-    },
-    discord: {
-      type: 'string',
-      title: 'Discord',
-      examples: ['Discord handle']
-    },
-    walletNetwork: {
-      type: 'string',
-      enum: enabledNetworks,
-      options: availableNetworks,
-      title: 'Treasury network'
-    },
-    walletAddress: {
-      type: 'string',
-      format: 'address',
-      title: 'Treasury address',
-      examples: ['0x0000…']
+const definition = computed(() => {
+  return {
+    type: 'object',
+    title: 'Space',
+    additionalProperties: false,
+    required: ['name', 'walletNetwork', 'walletAddress'],
+    properties: {
+      name: {
+        type: 'string',
+        title: 'Name',
+        minLength: 1,
+        examples: ['Space name']
+      },
+      description: {
+        type: 'string',
+        format: 'long',
+        title: 'About',
+        examples: ['Space description']
+      },
+      externalUrl: {
+        type: 'string',
+        title: 'Website',
+        examples: ['Website URL']
+      },
+      github: {
+        type: 'string',
+        title: 'GitHub',
+        examples: ['GitHub handle']
+      },
+      twitter: {
+        type: 'string',
+        title: 'Twitter',
+        examples: ['Twitter handle']
+      },
+      discord: {
+        type: 'string',
+        title: 'Discord',
+        examples: ['Discord handle']
+      },
+      walletNetwork: {
+        type: ['string', 'null'],
+        enum: [null, ...enabledNetworks],
+        options: [{ id: null, name: 'No treasury' }, ...availableNetworks],
+        title: 'Treasury network',
+        nullable: true
+      },
+      walletAddress: {
+        type: ['string', 'null'],
```

Comment:
I think type there should be conditional, otherwise it's possible to select network, without putting in address and it's valid.

Reply:
So we need to change initial walletAddress value to null then